src/components/Item/Item.tsx

```diff
@@ -223,26 +224,37 @@ function ItemMenuButton({
   );
 }
 
-export function draggableItemFactory({
-  items,
-  laneIndex,
-}: DraggableItemFactoryParams) {
-  return (
-    provided: DraggableProvided,
-    snapshot: DraggableStateSnapshot,
-    rubric: DraggableRubric
-  ) => {
-    const { boardModifiers, board } = React.useContext(KanbanContext);
-    const { view } = React.useContext(ObsidianContext);
-    const { query } = React.useContext(SearchContext);
-    const [isEditing, setIsEditing] = React.useState(false);
-
-    const itemIndex = rubric.source.index;
-    const item = items[itemIndex];
-    const lane = board.lanes[laneIndex];
+type DraggableItemState = {
+  isEditing: boolean;
+}
+
+export class DraggableItem extends React.PureComponent<DraggableItemProps, DraggableItemState> {
```

Comment:
@pjeby I don't mind using a class component here, but I believe this would essentially be the same as wrapping the old `DraggableItem` function you had with `React.memo`. Eg. `export const DraggableItem = React.memo((props: DraggableItemProps) => {...})`.

I generally prefer to use functional components where I can, as I've found them to be easier to refactor over time, and easier to test (when that becomes a thing 🤞). They do take more work to optimize, though, and class components are definitely the right choice when complex optimizations or instance states are needed.

Reply:
Does memo let you change shouldComponentUpdate?  The idea here is that items should ultimately ignore changes that are only to their co-ordinates (laneIndex/itemIndex).  I only didn't implement that part because I didn't want to hold you up on merging today.

Reply:
Yeah, you can pass in a second function: 

```
function MyComponent(props) {
  /* render using props */
}
function areEqual(prevProps, nextProps) {
  /*
  return true if passing nextProps to render would return
  the same result as passing prevProps to render,
  otherwise return false
  */
}
export default React.memo(MyComponent, areEqual);
```

Reply:
Nice.